### PR TITLE
feat: Fix integration test

### PR DESCRIPTION
**Pull request by toni-calvin**
In roll-up mode: 

image

In validium mode:

image

### CODE REVIEW

**Review by ilitteri — COMMENTED** [review]
Careful with the changes in `chain.toml` and `eth_sender.toml`, it seems that it was not formatted with `zk fmt`

File: infrastructure/zk/src/init.ts
Comment (ilitteri) [inline infrastructure/zk/src/init.ts:1]: Why is this change needed?

**Comment by jordibonet-lambdaclass** [comment]
> Careful with the changes in `chain.toml` and `eth_sender.toml`, it seems that it was not formatted with `zk fmt`

Fixed!

File: infrastructure/zk/src/init.ts
Comment (ilitteri) [inline infrastructure/zk/src/init.ts:1]: I think that `updateConfig` would be a better name for today's `configMode` function given that the function updates the config given the mode in which the stack is being initialized.

[the remark this answers is not in the document]
Comment (ilitteri) [reply infrastructure/zk/src/init.ts:1]: Now it's ok, but it would've been better to tackle the issue in a separate PR.

File: infrastructure/zk/src/init.ts [hunk 1]
```diff
@@ -138,6 +133,67 @@ export async function submoduleUpdate() {
     await utils.exec('git submodule update');
 }
 
+async function configMode(validiumMode: boolean) {
+    let envFileContent = fs.readFileSync(process.env.ENV_FILE!).toString();
+    envFileContent += `VALIDIUM_MODE=${validiumMode}\n`;
+    fs.writeFileSync(process.env.ENV_FILE!, envFileContent);
+    await announced(`Initializing in ${validiumMode ? 'Validium mode' : 'Roll-up mode'}`);
+
+    const chainPath = 'etc/env/base/chain.toml';
```
Comment (ilitteri) [inline infrastructure/zk/src/init.ts:142]: Let's also define a constant for this file name.

File: infrastructure/zk/src/init.ts [hunk 2]
```diff
@@ -138,6 +133,67 @@ export async function submoduleUpdate() {
...
+            }
+        }
+    }
+    chainContent = lines.join('\n');
+    fs.writeFileSync(chainPath, chainContent);
+
+    await announced(`The parameters have been updated in the ${chainPath} file.`);
+
+    const ethSenderPath = 'etc/env/base/eth_sender.toml';
+    const enforcedGasPrice = 'internal_enforced_l1_gas_price';
```
Comment (ilitteri) [inline infrastructure/zk/src/init.ts:167]: Let's keep names consistent

```suggestion
    const internalEnforcedL1GasPrice = 'internal_enforced_l1_gas_price';
```

File: infrastructure/zk/src/init.ts [hunk 3]
```diff
@@ -138,6 +133,67 @@ export async function submoduleUpdate() {
...
+    const newValue = 45_000_000_000;
```
Comment (ilitteri) [inline infrastructure/zk/src/init.ts:168]: Define a constant for this value.

File: infrastructure/zk/src/init.ts [hunk 4]
```diff
@@ -138,6 +133,67 @@ export async function submoduleUpdate() {
...
+            break;
+        }
+    }
+
+    if (!found) {
+        if (validiumMode) {
+            linesEthSender.pop();
+            linesEthSender.push(`${enforcedGasPrice}=${newValue}\n`);
+        }
+    }
```
Comment (ilitteri) [inline infrastructure/zk/src/init.ts:189]:
```suggestion
    if (!found && validiumMode) {
        linesEthSender.pop();
        linesEthSender.push(`${enforcedGasPrice}=${newValue}\n`);
    }
```

File: infrastructure/zk/src/init.ts [hunk 5]
```diff
@@ -138,6 +133,67 @@ export async function submoduleUpdate() {
     await utils.exec('git submodule update');
 }
 
+async function configMode(validiumMode: boolean) {
```
Comment (ilitteri) [inline infrastructure/zk/src/init.ts:136]: We can abstract the logic that loads and updates both `chain.toml` and `eth_sender.toml` files into different functions (`updateChainConfig`, and `updateEthSenderConfig`) to improve the code's readability.

**Review by ilitteri — CHANGES_REQUESTED** [review]
In general, the changes made for making the tests pass are very clear, good work. Left some comments on that but very few ones.

The changes regarding #96 are also very good and what I expected. I left some comments regarding the implementation. As a TLDR: for this to be merged we need to get rid of "magic numbers" and hardcoded file paths and define some constants, also refactor the code a little bit (I proposed some of them).

File: infrastructure/zk/src/init.ts [hunk 6]
```diff
@@ -138,6 +136,64 @@ export async function submoduleUpdate() {
...
+    updateConfigFile(CHAIN_CONFIG_PATH, modeConstantValues);
+}
+function updateEthSenderConfig(validiumMode: boolean) {
+    const modeConstantValues = {
+        internal_enforced_l1_gas_price: validiumMode
+            ? constants.VALIDIUM_ENFORCED_L1_GAS_PRICE
+            : constants.ROLLUP_ENFORCED_L1_GAS_PRICE
+    };
+    updateConfigFile(ETH_SENDER_PATH, modeConstantValues);
+}
```
Comment (ilitteri) [inline infrastructure/zk/src/init.ts:187]: Probably missed this on Friday (long day), but I have a comment here, the `internal_enforced_l1_gas_price` is a variable we override with `45_000_000_000` because in Ethereum's mainnet this is the value for the gas price. We set it like that to see better results when comparing Validium mode vs Rollup mode. By this, I mean that we need to get rid of updating the `eth_sender_config` with this (it is ok to have it hardcoded in our PRs but then this will be reverted in the PR to `main`).

File: etc/env/base/eth_sender.toml [hunk 7]
```diff
@@ -62,4 +62,4 @@ pricing_formula_parameter_b=1.0005
 internal_l1_pricing_multiplier=0.8
 # Node polling period in seconds.
 poll_period=5
-internal_enforced_l1_gas_price=45_000_000_000
+internal_enforced_l1_gas_price=45000000000
```
[the remark this answers is not in the document]
Comment (ilitteri) [reply etc/env/base/eth_sender.toml:65]: Unresolving this, I still see the diff.

**Review by ilitteri — CHANGES_REQUESTED** [review]
Great job refactoring the code! One more thing that I missed last week and we can merge

File: infrastructure/zk/src/init.ts (hunk 6 above)
Comment (ilitteri) [reply infrastructure/zk/src/init.ts:187]: Also, it is the same for both modes.

File: infrastructure/zk/src/init.ts (hunk 4 above)
Comment (toni-calvin) [reply infrastructure/zk/src/init.ts]: Code changed!

File: infrastructure/zk/src/init.ts (hunk 6 above)
Comment (toni-calvin) [reply infrastructure/zk/src/init.ts]: 👍

File: core/tests/ts-integration/src/helpers.ts [hunk 8]
```diff
@@ -105,3 +105,25 @@ export async function scaledGasPrice(wallet: ethers.Wallet | zksync.Wallet): Pro
     // Increase by 40%.
     return gasPrice.mul(140).div(100);
 }
+
+/**
+ * Returns if it is runnning in Validium Mode.
+ *
+ * @returns Boolean that indicates whether it is Validium mode.
+ */
+export async function isValidium(): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (jordibonet-lambdaclass) [reply core/tests/ts-integration/src/helpers.ts:114]: We can't because they are different processes. It's the first thing we've tried.

[the remark this answers is not in the document]
Comment (ilitteri) [reply core/tests/ts-integration/src/helpers.ts:114]: Got it!

**Comment by jordibonet-lambdaclass** [comment]
We have changed the COMPUTE_OVERHEAD to be changeable because it was not passing the tests in Rollup, it was one of the first fixes we made (it fixes a bug in the web3 suite, from the gas difference).

We have also optimized the "updateConfigFile()" function to be of complexity O(n+m) instead of O(m·n), where "m" is the number of parameters and "n" is the number of lines in the configuration file.
